searchcore/src/tests/proton/matching/matching_stats_test.cpp

```diff
@@ -274,7 +274,29 @@ TEST("requireThatSoftDoomFacorIsComputedCorrectly") {
     stats.updatesoftDoomFactor(1.0, 0.5, 10.0);      // Prevent changes above 10%
     EXPECT_EQUAL(1ul, stats.softDoomed());
     EXPECT_EQUAL(0.396, stats.softDoomFactor());
+}
 
+TEST("requireThatSoftDoomFacorIsComputedCorrectlyForUpAdjustment") {
+    MatchingStats stats;
+    EXPECT_EQUAL(0ul, stats.softDoomed());
+    EXPECT_EQUAL(0.5, stats.softDoomFactor());
+    stats.softDoomed(1);
+    stats.updatesoftDoomFactor(1.0, 0.9, 0.1);
+    EXPECT_EQUAL(1ul, stats.softDoomed());
+    EXPECT_EQUAL(0.508, stats.softDoomFactor());
+    stats.updatesoftDoomFactor(1.0, 0.9, 0.1);
+    EXPECT_EQUAL(1ul, stats.softDoomed());
+    EXPECT_EQUAL(0.516, stats.softDoomFactor());
+    stats.updatesoftDoomFactor(0.0009, 0.9, 0.1);   // hard limits less than 1ms should be ignored
+    EXPECT_EQUAL(1ul, stats.softDoomed());
+    EXPECT_EQUAL(0.516, stats.softDoomFactor());
+    stats.updatesoftDoomFactor(1.0, 0.9, 0.1);   // soft limits less than 1ms should be ignored
```

Comment:
This seems to be the same as line 287 and is not ignored. softDoomFactor should be 0.516 instead of 0.524.

Reply:
softlimit 0.9 should be changed to 0.0009?